sentry-android-core/src/main/java/io/sentry/android/core/ViewHierarchyEventProcessor.java

```diff
@@ -73,39 +71,30 @@ public ViewHierarchyEventProcessor(final @NotNull SentryAndroidOptions options)
 
     try {
       final @NotNull ViewHierarchy viewHierarchy = snapshotViewHierarchy(decorView);
-      final @NotNull Future<?> future =
-          options
-              .getExecutorService()
-              .submit(
-                  () -> {
-                    try {
-                      serializeViewHierarchy(viewHierarchy, hint);
-                    } catch (Throwable e) {
-                      options
-                          .getLogger()
-                          .log(SentryLevel.ERROR, "Failed trying to serialize view hierarchy.", e);
-                    }
-                  });
-      future.get(TIMEOUT_PROCESSING_MILLIS, TimeUnit.MILLISECONDS);
+      attachViewHierarchy(viewHierarchy, hint);
     } catch (Throwable t) {
       options.getLogger().log(SentryLevel.ERROR, "Failed to process view hierarchy.", t);
     }
     return event;
   }
 
-  private void serializeViewHierarchy(@NonNull ViewHierarchy viewHierarchy, @NonNull Hint hint) {
-    try {
-      try (final ByteArrayOutputStream stream = new ByteArrayOutputStream();
-          final Writer writer = new BufferedWriter(new OutputStreamWriter(stream, UTF_8))) {
-
-        options.getSerializer().serialize(viewHierarchy, writer);
-
-        final Attachment attachment = Attachment.fromViewHierarchy(stream.toByteArray());
-        hint.setViewHierarchy(attachment);
-      }
-    } catch (Throwable t) {
-      options.getLogger().log(SentryLevel.ERROR, "Could not snapshot ViewHierarchy", t);
-    }
+  private void attachViewHierarchy(@NonNull ViewHierarchy viewHierarchy, @NonNull Hint hint) {
+    hint.setViewHierarchy(
+        Attachment.fromViewHierarchy(
```

Comment:
M: Since `Attachment.fromViewHierarchy` is a public API, why don't we allow `Attachment.fromViewHierarchy(viewHierarchy)` instead of the callable? so the user does not need to handle all the serialization bits of a format they don't understand, but rather creating the `ViewHierarchy` object, a much user-friendlier API, that's how it is on Dart.